kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/shared/basetests/KieServerBaseIntegrationTest.java

```diff
@@ -0,0 +1,207 @@
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.KieServicesFactory;
+import org.kie.server.controller.api.model.KieServerInstance;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.integrationtests.controller.client.KieServerMgmtControllerClient;
+import org.kie.server.integrationtests.config.TestConfig;
+import org.kie.server.integrationtests.controller.client.KieServerControllerClient;
+import org.kie.server.integrationtests.shared.KieControllerExecutor;
+import org.kie.server.integrationtests.shared.KieServerExecutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class KieServerBaseIntegrationTest {
+
+    protected static Logger logger = LoggerFactory.getLogger(KieServerBaseIntegrationTest.class);
+
+    protected KieServicesClient client;
+    protected static KieControllerExecutor controller;
+    protected static KieServerExecutor server;
+
+    @BeforeClass
+    public static void setupClass() throws Exception {
+        if (TestConfig.isLocalServer()) {
+            System.setProperty(Context.INITIAL_CONTEXT_FACTORY, "bitronix.tm.jndi.BitronixInitialContextFactory");
+            controller = new KieControllerExecutor();
+            controller.startKieController();
+            server = new KieServerExecutor();
+            server.startKieServer();
+        }
+        setupCustomSettingsXml();
+    }
+
+    private static void setupCustomSettingsXml() {
+        if (!TestConfig.isLocalServer()) {
+            String deploymentSettings = TestConfig.getKieClientDeploymentSettings();
+
+            if (deploymentSettings == null) {
+                deploymentSettings = ClassLoader.class.getResource("/kie-server-testing-client-deployment-settings.xml").getFile();
+            }
+
+            System.setProperty(KieServerConstants.CFG_KIE_MVN_SETTINGS, deploymentSettings);
+        }
+    }
+
+    @Before
+    public void setup() throws Exception {
+        startClient();
+        disposeAllContainers();
+        disposeAllServerInstances();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        if (TestConfig.isLocalServer()) {
+            server.stopKieServer();
+            controller.stopKieController();
+            System.clearProperty(Context.INITIAL_CONTEXT_FACTORY);
+        }
+    }
+
+    protected void disposeAllContainers() {
+        ServiceResponse<KieContainerResourceList> response = client.listContainers();
+        Assert.assertEquals(ServiceResponse.ResponseType.SUCCESS, response.getType());
+        List<KieContainerResource> containers = response.getResult().getContainers();
+        if (containers != null) {
+            for (KieContainerResource container : containers) {
+                client.disposeContainer(container.getContainerId());
+            }
+        }
+    }
+
+    protected void disposeAllServerInstances() throws Exception {
```

Comment:
It seems - at least to me - that this method combines several things - deletes instances and server templates. Maybe it would be better to decompose or rename it. Wdyt?